study/src/test/java/study/IOStreamTest.java

```diff
@@ -128,7 +142,19 @@ class InputStream_학습_테스트 {
              * todo
              * inputStream에서 바이트로 반환한 값을 문자열로 어떻게 바꿀까?
              */
-            final String actual = "";
+
+            byte[] readBytes = new byte[4];
+
+            int i = 0;
+            while (true) {
+                int read = inputStream.read();
+                if (read == -1) {
+                    break;
+                }
+                readBytes[i] = (byte) read;
+                i++;
+            }
+            final String actual = new String(readBytes);
```

Comment:
이 코드는 `read` 메서드가 `-1` 을 리턴하는걸 눈으로 확인해보고 싶어서 의도적으로 반복문으로 작성하게 되었습니다!

처음에 "왜 `read`가 byte가 아닌 int를 리턴하는거지?" 라는 궁금증에서 코드의 구현을 찾아보게 되었습니다. 

>The value byte is returned as an int in the range 0 to 255. If no byte is available because the end of the stream has been reached, the value -1 is returned. 

기본적으로 read는 스트림에서 바이트를 읽어 0 ~ 255의 값을 리턴해주고 더이상 읽을게 없을때는 -1을 리턴해주도록 구현되었습니다. 이때 -1 ~ 255를 담기에는 byte는 부족하기 때문에 read가 byte가 아닌 int를 리턴하고 불편하게 형변환을 해줘야 하는것을 알게되었습니다!

그래서 이 코드는 유지하고 싶습니다!

Reply:
좋은 정보 감사합니다!! 로키의 학구열이 정말 대단하네요 👍

학습용 코드인만큼 로키가 원하는 대로 두어도 좋을 것 같습니다!